util/csrf.py

```diff
@@ -13,8 +13,12 @@ def check_csrf_protection():
     on a per-view basis, maybe with a decorator.
     """
     if request.method == 'POST':
-        token = session.pop('_csrf_token', None)
-        if not token or token != request.form.get('_csrf_token'):
+        token = session.get('_csrf_token')
+        payload = request.form.get('_csrf_token')
+        # request form of csrf token is returned as a string of a bytestring, slicing to just retrieve the bytestring
+        # Not sure how else to do this :/
```

Comment:
That's because it's already set like that:
```html
<input name="_csrf_token" type="hidden" value="b&#39;f6c75f2e3c191a162335e5ad99b8089d&#39;">
```
This is a bug in the app related to Python 3, we'll need to fix how we set the csrf token.

Reply:
In line 27, we have this code:
```py
session['_csrf_token'] = binascii.hexlify(os.urandom(16))
```

`hexlify` returrns a bytestring. The fix is to decode it:

```py
session['_csrf_token'] = binascii.hexlify(os.urandom(16)).decode('ascii')
```

Reply:
I wasn't sure if the csrf token needed to be of type `bytes`. Will fix!